test/bntest.c

```diff
@@ -185,6 +185,7 @@ static int test_sub()
 {
     BIGNUM *a, *b, *c;
     int i;
+    int st = 0;
 
     a = BN_new();
```

Comment:
or do all the BN_new calls and then a single if to test if a b or c is NULL.
Or, for tests, do not worry very much.  The test will crash and that's useful :)